crates/swc_ecma_parser/src/parser/typescript.rs

```diff
@@ -398,16 +398,16 @@ impl<I: Tokens> Parser<I> {
         )? {
             match modifer {
                 "const" => {
+                    is_const = true;
```

Comment:
It's better to reflect the state of the code in the AST even when a diagnostic is being emitted.

Reply:
I agree, thank you.